### PR TITLE
Get version from started.json.

### PR DESCRIPTION
**Pull request by david-mcmahon**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30385)

### CODE REVIEW

**Comment by spxtr** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385#-KOpomOxCBIPVN5ceYaj:-KOpomOxCBIPVN5ceYak:b-3dpt80) (no related file):*
I thought the point of this was to get the version *for* started.json.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385)*

**Comment by david-mcmahon** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385#-KOpomOxCBIPVN5ceYaj:-KOpwr43_RB7D0OWv9dx:batm5u) (no related file):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> I thought the point of this was to get the version *for* started.json.
</details>
find_version() is used in the started and finished contexts.  Now find_version() will find version in the appropriate way for the started context in the first two conditions ultimately falling through to get it from started.json itself in the finished context.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385)*

File: hack/jenkins/upload-to-gcs.sh [hunk 1]
```diff
@@ -94,6 +94,10 @@ function find_version() {
     source "hack/lib/version.sh"
     kube::version::get_version_vars
     echo "${KUBE_GIT_VERSION-}"
+  else
+    # Last resort from the started.json
+    gsutil cat ${gcs_build_path}/started.json 2>/dev/null |\
+     jq -r '.version|values'
```
Comment (ixdy) [inline hack/jenkins/upload-to-gcs.sh]: I'm not sure we have `jq` installed on the Jenkins VMs.

**Comment by ixdy** [comment]
Can you link to a run where this still failed? I'd like to understand what's wrong.

**Comment by ixdy** [comment]
OK, I finally realized why this doesn't work (writing here again for historical purposes) - for e2e runs, we extract the kubernetes tarball inside the Docker container. As such, when we get to the final upload-to-gcs bit, which runs outside of this Docker context, the extracted artifacts no longer exist. Hence no version file, anywhere.

**Comment by david-mcmahon** [comment]
Good ol' trusty sed to the rescue.  PTAL.

File: hack/jenkins/upload-to-gcs.sh [hunk 2]
```diff
@@ -94,6 +94,10 @@ function find_version() {
     source "hack/lib/version.sh"
     kube::version::get_version_vars
     echo "${KUBE_GIT_VERSION-}"
+  else
+    # Last resort from the started.json
+    gsutil cat ${gcs_build_path}/started.json 2>/dev/null |\
```
Comment (ixdy) [inline hack/jenkins/upload-to-gcs.sh]: "convenient" that gsutil doesn't exit nonzero if the file doesn't exist.

**Comment by ixdy** [comment]
LGTM

File: hack/jenkins/upload-to-gcs.sh (hunk 2 above)
Comment (david-mcmahon) [inline hack/jenkins/upload-to-gcs.sh]: Either way it's further piped into sed which is all 'set -e' cares about anyway.

Comment (ixdy) [inline hack/jenkins/upload-to-gcs.sh]: but we also set pipefail.

Comment (david-mcmahon) [inline hack/jenkins/upload-to-gcs.sh]: Ah yes. We'll convenient then.

**Comment by david-mcmahon** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[hack/jenkins/upload-to-gcs.sh, line 100 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385#-KOpeklE8b1-1allMaBE-r1-100:-KOujzawsa7sKsaBWeeZ:bvhz6zi) ([raw file](https://github.com/kubernetes/kubernetes/blob/83e290afd5b60643e1a5e895ca895243a3e397b2/hack/jenkins/upload-to-gcs.sh#L100)):*
<details><summary><i>Previously, ixdy (Jeff Grafton) wrote…</i></summary>
> I'm not sure we have `jq` installed on the Jenkins VMs.
</details>
Yes, fixed with sed. 

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385)*

**Comment by david-mcmahon** [comment]
Reviewable is holding up this PR because comments in the discussion we had there are not marked as resolved. Ugh.  @spxtr @ixdy can you go resolve those?

**Comment by ixdy** [comment]
@david-mcmahon sure, though it looks like mergebot doesn't care about that (at least not yet)

**Comment by ixdy** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385#-:-KOqRetrgpY7EQ_0ZY1k:bnfp4nl)*

**Comment by david-mcmahon** [comment]
oh ok, not gating then.  I just saw the yellow dot and assumed..

**Comment by spxtr** [comment]
Reviewed 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30385)*

**Pull request merged**